src/bindings/g_object/object.cr

```diff
@@ -67,6 +73,296 @@ module GObject
 
         # :nodoc:
         def self._class_init(klass : Pointer(LibGObject::TypeClass), user_data : Pointer(Void)) : Nil
+          {% verbatim do %}
+            {% begin %}
+              {% instance_vars = @type.instance_vars.select(&.annotation(GObject::Property)) %}
+
+              @@_g_param_specs = Pointer(LibGObject::ParamSpec*).malloc({{ instance_vars.size }})
+              {% for var, i in instance_vars %}
+                {% property = var.annotation(GObject::Property) %}
+                name = {{ var.name.gsub(/\_/, "-").stringify }}.to_unsafe
+                nick = {{ property["nick"] }}.try(&.to_unsafe) || Pointer(LibC::Char).null
+                blurb = {{ property["blurb"] }}.try(&.to_unsafe) || Pointer(LibC::Char).null
+                {% other_args = property.named_args.to_a.reject { |arg| ["nick", "blurb"].includes?(arg[0].stringify) } %}
+
+                {% has_getter = @type.has_method?(var.name.stringify) %}
+                {% has_setter = @type.has_method?("#{var.name}=") %}
+                {% raise "GObject properties need to have a getter and/or a setter" unless has_getter || has_setter %}
+
+                flags = GObject::ParamFlags::StaticName | GObject::ParamFlags::StaticBlurb | GObject::ParamFlags::ExplicitNotify
+                flags |= GObject::ParamFlags::Deprecated unless {{ !!var.annotation(Deprecated) }}
+                flags |= GObject::ParamFlags::Readable if {{ has_getter }}
+                flags |= GObject::ParamFlags::Writable if {{ has_setter }}
+
+                # Finally register the type to GLib.
+                # The given varible name has its underscores converted to dashes.
+                pspec = GObject.create_param_spec({{ var.type }}, name, nick, blurb, flags, {{ other_args.map { |tuple| "#{tuple[0]}: #{tuple[1]}".id }.splat }})
+                @@_g_param_specs[{{ i }}] = pspec.as(LibGObject::ParamSpec*)
+                LibGObject.g_object_class_install_property(klass, {{ i + 1 }}, pspec)
+              {% end %}
+            {% end %}
+          {% end %}
+        end
+
+        # :nodoc:
+        @[GObject::Virtual(name: "get_property", unsafe: true)]
```

Comment:
Here

Reply:
Weird... I got your last version with the merge from master from minutes ago... but after the rebase the git auto-merge must have messed up thing in object.cr file.

And now I ask what was really merged :rofl: , anyway... I reviewed the diff and it looks the same.